### PR TITLE
Change how we declare functions to encode and decode primitive arrays 

### PR DESCRIPTION
**Pull request by swankjesse**

### CODE REVIEW

**Comment by swankjesse** [comment]
Note this is not backwards-compatible for anyone using the optimized arrays that were introduced in 4.5.6.

File: wire-runtime/api/wire-runtime.api [hunk 1]
```diff
@@ -32,22 +32,6 @@ public final class com/squareup/wire/AnyMessage$Companion {
 	public final fun pack (Lcom/squareup/wire/Message;)Lcom/squareup/wire/AnyMessage;
 }
 
-public final class com/squareup/wire/DoubleProtoAdapter : com/squareup/wire/ProtoAdapter {
```
Comment (oldergod) [inline wire-runtime/api/wire-runtime.api:35]: Is this like picasso where CI would fail if we don't manually update ourselves those files?

**Review by oldergod — APPROVED** [review]

File: wire-runtime/api/wire-runtime.api (hunk 1 above)
Comment (oldergod) [reply wire-runtime/api/wire-runtime.api:35]: pretty sweet https://github.com/Kotlin/binary-compatibility-validator#tasks

Comment (swankjesse) [reply wire-runtime/api/wire-runtime.api:35]: Exactly!

**Pull request merged**